lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -366,6 +366,33 @@ sub register ($self, $app, $config) {
 
             $c->res->headers->links($links);
         });
+
+    $app->helper(
+        regex_problem => sub ($c, $regexes, $context = undef) {
+            my $regex_problem;
+            state %problematic_regexes;
+            for my $regex_string (@$regexes) {
+                # check whether the regex is already known to be problematic
+                # note: This is not an optimization. It is required because some warnings are only logged for the first
+                #       time they are encountered.
+                last if $regex_problem = $problematic_regexes{$regex_string};
+                # record warnings as some problems are not fatal
+                # note: We should not leave those warnings unhandled as they would end up in the log.
+                #       An example for such a warning is "$* matches null string many times in regex".
+                local $SIG{__WARN__} = sub ($warning, @) { $regex_problem = $warning };
```

Comment:
Instead of using a signal handler I would make warnings fatal in this block:
```
use warnings FATAL => "regexp";
```
Then the eval will die also in those cases.

Reply:
I didn't know that was possible. I suppose I'll do that then.